analytics.js

```diff
@@ -1336,21 +1336,34 @@ const eventsPushedAlready =
   !!window.rudderanalytics &&
   window.rudderanalytics.push == Array.prototype.push;
 
-const argumentsArray = window.rudderanalytics;
-
-while (argumentsArray && argumentsArray[0] && argumentsArray[0][0] !== "load") {
-  argumentsArray.shift();
+const argumentsArray = window.rudderanalytics || [];
```

Comment:
```suggestion
const argumentsArray = window.rudderanalytics;
```

Reply:
If someone only loads the CDN url, there will be no `argumentsArray`. In that case from Ln 1348 and 1366 they will get error `length of undefined`.

Reply:
Addressed

Reply:
It wasn’t there in the previous implementation. We added that because in the case of undefined argumentsArray users will get an error. 
The undefined case is handled and if the users use our loading code snippet as mentioned in our docs there will not be any case of `argumentsArray` being anything else other than an array. 
Let’s not add the extra checking that might lead to some other issue that at this moment we can not think of.

Reply:
Addressed